packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/__tests__/quanticRecentQueriesList.test.js

```diff
@@ -0,0 +1,444 @@
+);
+jest.mock(
+  '@salesforce/label/c.quantic_EmptyRecentQueriesListLabel',
+  () => ({default: 'Your recent searches will appear here.'}),
+  {
+    virtual: true,
+  }
+);
+jest.mock(
+  '@salesforce/label/c.quantic_Expand',
+  () => ({default: 'Expand Recent queries'}),
+  {
+    virtual: true,
+  }
+);
+jest.mock(
+  '@salesforce/label/c.quantic_Collapse',
+  () => ({default: 'Collapse Recent queries'}),
+  {
+    virtual: true,
+  }
+);
+
+const defaultRecentQueriesTitle = 'Recent Queries';
+const defaultQueries = [];
+const defaultMaxLength = 10;
+let isInitialized = false;
+let updateRecentQueriesState;
+
+const initialRecentQueriesListState = {
+  queries: defaultQueries,
+  maxLength: defaultMaxLength,
+};
+let recentQueriesListState = initialRecentQueriesListState;
+
+const functionMocks = {
+  buildRecentQueriesList: jest.fn(() => ({
+    state: recentQueriesListState,
+    subscribe: functionMocks.recentQueriesListSubscriber,
+    executeRecentQuery: functionMocks.executeRecentQuery,
+    clear: functionMocks.clear,
+    updateRecentQueries: functionMocks.updateRecentQueries,
+  })),
+  recentQueriesListSubscriber: jest.fn((callback) => {
+    updateRecentQueriesState = async () => {
+      callback();
+      await flushPromises();
+    };
+    return () => {};
+  }),
+  recentQueriesListUnsubscriber: jest.fn(),
+  executeRecentQuery: jest.fn(),
+  clear: jest.fn(),
+  updateRecentQueries: jest.fn(),
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  label: 'Recent Queries',
+  hideWhenEmpty: false,
+  isCollapsed: false,
+};
+
+const selectors = {
+  initializationError: 'c-quantic-component-error',
+  placeholder: 'c-quantic-placeholder',
+  quanticCardContainer: 'c-quantic-card-container',
+  recentQueryItem: '[data-testid="recent-query-item"]',
+  recentQueryItemText: '[data-testid="recent-query-text"]',
+  toggleButton: '[data-testid="action-button"]',
+};
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticRecentQueriesList,
+  'c-quantic-recent-queries-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildRecentQueriesList: functionMocks.buildRecentQueriesList,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticRecentQueriesList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticRecentQueriesList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-recent-queries-list', () => {
+  afterEach(() => {
+    cleanup();
+    recentQueriesListState = initialRecentQueriesListState;
+    isInitialized = false;
+  });
```

Comment:
If you're playing with mocks and spy in this file, I would take the time to add here a

`jest.resetAllMocks()`

to the afterEach()

That way you are sure you don't have mocks leaking into other tests.

Reply:
I tought it was already in the cleanup, apparently not

Reply:
its actually breaking most tests, I dont think we already do `jest.clearAllMocks();` in the cleanup() method